### PR TITLE
Feature/edit libraries

### PR DESCRIPTION
**Pull request by 0x64746b**
This PR makes the `expandable_toggle_button` visible for `SeafRepo` items and binds it to a new "bottom sheet", which makes it possible to `rename` and `delete` repos.

### CODE REVIEW

**Comment by Logan676** [comment]
#557

**Comment by 0x64746b** [comment]
@Logan676 Can you read the `gradlew` output? I don't understand where the problem is and cannot reproduce it locally:

```
➜  seadroid git:(feature/edit-libraries) ✗ ./gradlew clean assemble
Incremental java compilation is an incubating feature.
:app:clean
:app:preBuild UP-TO-DATE
:app:preDebugBuild UP-TO-DATE
:app:checkDebugManifest
:app:preReleaseBuild UP-TO-DATE
   [...]
  Copying resources from program directory [/home/dtk/Development/seadroid/app/build/intermediates/classes/release] (filtered)
Printing classes to [/home/dtk/Development/seadroid/app/build/outputs/mapping/release/dump.txt]...
:app:transformClassesWithDexForRelease
:app:mergeReleaseJniLibFolders
:app:transformNative_libsWithMergeJniLibsForRelease
:app:validateReleaseSigning
:app:packageRelease
:app:zipalignRelease
:app:assembleRelease
:app:assemble

BUILD SUCCESSFUL

Total time: 1 mins 18.901 secs

This build could be faster, please consider using the Gradle Daemon: https://docs.gradle.org/2.10/userguide/gradle_daemon.html
➜  seadroid git:(feature/edit-libraries) ✗ echo $?
0
➜  seadroid git:(feature/edit-libraries) ✗ 
```

**Comment by Logan676** [comment]
Do you mean that travis ci build failed problem? If that's the case, ignore it, I will look into it.

**Comment by 0x64746b** [comment]
> ignore it, I will look into it.

Great, thx.

File: app/src/main/java/com/seafile/seadroid2/ui/dialog/DeleteLibDialog.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+
+import com.seafile.seadroid2.R;
+import com.seafile.seadroid2.SeafException;
+import com.seafile.seadroid2.account.Account;
+import com.seafile.seadroid2.data.DataManager;
+
+class DeleteLibTask extends TaskDialog.Task {
```
Comment (Logan676) [inline app/src/main/java/com/seafile/seadroid2/ui/dialog/DeleteLibDialog.java]: DeleteRepoTask is better I think.

File: app/src/main/java/com/seafile/seadroid2/ui/dialog/DeleteLibDialog.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        if (mDataManager == null) {
+            mDataManager = new DataManager(mAccount);
+        }
+
+        return mDataManager;
+    }
+
+    @Override
+    protected View createDialogContentView(LayoutInflater inflater, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.dialog_delete_lib, null);
```
Comment (Logan676) [inline app/src/main/java/com/seafile/seadroid2/ui/dialog/DeleteLibDialog.java]: dialog_delete_repo is better

File: app/src/main/java/com/seafile/seadroid2/ui/activity/BrowserActivity.java [hunk 3]
```diff
@@ -1822,6 +1824,22 @@ public void onFailure(SeafException err) {
         fetchFileDialog.show(getSupportFragmentManager(), OPEN_FILE_DIALOG_FRAGMENT_TAG);
     }
 
+    public void deleteLib(String repoID) {
```
Comment (Logan676) [inline app/src/main/java/com/seafile/seadroid2/ui/activity/BrowserActivity.java]: deleteRepo is better

**Comment by 0x64746b** [comment]
@Logan676 It feels a bit like you looked at an old version; then again I don't think I even create this PR before efbd44e6d91871bf5d7ca33d91a39c82c8087b3f :/

**Comment by Logan676** [comment]
Right, I have looked at an old version. Actually I reviewed the patch commit by commit, so I didn\`t realize that you already refactored the code at that moment. Now everything is alright.

**Pull request merged**